core/controller/src/main/scala/whisk/core/controller/Actions.scala

```diff
@@ -702,19 +702,19 @@ trait WhiskActionsApi extends WhiskCollectionAPI {
      * @param sequences the sequences encountered during inlining
      * @return a successful future in case the conditions were met or a failed fugure with either too many actions or recursion found
      */
-    private def inlineComponentsAndCountAtomicActions(atomicActionsCnt: Int, actionsToInline: List[String], sequences: List[String])(
+    private def inlineComponentsAndCountAtomicActions(atomicActionsCnt: Int, actionsToInline: List[FullyQualifiedEntityName], sequences: List[FullyQualifiedEntityName])(
         implicit transid: TransactionId): Future[Boolean] = {
         if (atomicActionsCnt > whiskConfig.actionSequenceLimit.toInt)
             Future.failed(new TooManyActionsInSequence())
         else {
             actionsToInline match {
                 case action :: restActions =>
-                    val docid = DocId(action)
+                    val docid = DocId(action.toString)
```

Comment:
Not entirely sure (without digging deeper) what the type of action is here but I think it's better to avoid toString and instead add a toDocId method on action.